common/src/main/java/com/microsoft/identity/common/internal/cache/ADALTokenCacheItem.java

```diff
@@ -83,9 +80,10 @@
         mSpeRing = tokenCacheItem.getSpeRing();
     }
 
-    ADALTokenCacheItem(OAuth2Strategy strategy, AuthorizationRequest request, TokenResponse response) {
-
-        Account account = strategy.createAccount(response);
+    ADALTokenCacheItem(final AzureActiveDirectoryOAuth2Strategy strategy,
+                       final AzureActiveDirectoryAuthorizationRequest request,
+                       final AzureActiveDirectoryTokenResponse response) {
+        AzureActiveDirectoryAccount account = strategy.createAccount(response);
         String issuerCacheIdentifier = strategy.getIssuerCacheIdentifier(request);
         AccessToken accessToken = strategy.getAccessTokenFromResponse(response);
```

Comment:
Should it be AzureActiveDirectoryAccessToken?

Reply:
Because the strategy is specified.

Reply:
It is an `AzureActiveDirectoryAccessToken` "under the hood" -- the left-side declaration is more high-level since this class doesn't do anything requiring the subclass. If it needs to do something else in the future, you can update that declaration and everything will work as expected